src/message/mailbox/parsers/rfc2822.rs

```diff
@@ -0,0 +1,249 @@
+//! Partial parsers implementation of [RFC2822]: Internet Message
+//! Format.
+//!
+//! [RFC2822]: https://datatracker.ietf.org/doc/html/rfc2822
+
+use chumsky::{error::Cheap, prelude::*};
+
+use super::{rfc2234, rfc5336};
+
+// 3.2.1. Primitive Tokens
+// https://datatracker.ietf.org/doc/html/rfc2822#section-3.2.1
+
+// NO-WS-CTL       =       %d1-8 /         ; US-ASCII control characters
+//                         %d11 /          ;  that do not include the
+//                         %d12 /          ;  carriage return, line feed,
+//                         %d14-31 /       ;  and white space characters
+//                         %d127
+fn no_ws_ctl() -> impl Parser<char, char, Error = Cheap<char>> {
+    filter(|c| matches!(u32::from(*c), 1..=8 | 11 | 12 | 14..=31 | 127))
+}
+
+// text            =       %d1-9 /         ; Characters excluding CR and LF
+//                         %d11 /
+//                         %d12 /
+//                         %d14-127 /
+//                         obs-text
+fn text() -> impl Parser<char, char, Error = Cheap<char>> {
+    filter(|c| matches!(u32::from(*c), 1..=9 | 11 | 12 | 14..=127))
+}
+
+// 3.2.2. Quoted characters
+// https://datatracker.ietf.org/doc/html/rfc2822#section-3.2.2
+
+// quoted-pair     =       ("\" text) / obs-qp
+fn quoted_pair() -> impl Parser<char, char, Error = Cheap<char>> {
+    just('\\').ignore_then(text())
+}
+
+// 3.2.3. Folding white space and comments
+// https://datatracker.ietf.org/doc/html/rfc2822#section-3.2.3
+
+// FWS             =       ([*WSP CRLF] 1*WSP) /   ; Folding white space
+//                         obs-FWS
+pub fn fws() -> impl Parser<char, Vec<char>, Error = Cheap<char>> {
+    rfc2234::wsp()
+        .ignored()
+        .repeated()
+        .map(|wsp| if wsp.is_empty() { vec![] } else { vec![' '] })
+}
+
+// CFWS            =       *([FWS] comment) (([FWS] comment) / FWS)
+pub fn cfws() -> impl Parser<char, Vec<char>, Error = Cheap<char>> {
+    // NOTE: comment should not be present, so cfws is the same as
```

Comment:
While not supported comments is fine for my use case I think you can use lettre to parse any email, not just emails generated by lettre. So I think we should support comments. IIUC if we just return `Some(' ')` and use the `Cheap` error it should be minimally expensive to support comments.

Reply:
I can put back the comment support, but IMHO it is a waste of time and perfs:

* Comments were not supported before this PR, and no one mention anything about comments before, I doubt that adding it back we help anyone. We can still add it back in the future if needed (with a better implem).
* Perfs are really worse with

Reply:
I don't think it is critical. They weren't supported before so I wouldn't complain about not adding them. However can we update the comment to say something like "comments not currently supported" rather than "should not be present"?

Reply:
:heavy_check_mark: